src/QsCompiler/Core/SyntaxGenerator.fs

```diff
@@ -204,13 +216,13 @@ module SyntaxGenerator =
     let CallNonGeneric (lhs: TypedExpression, rhs: TypedExpression) =
         let kind = CallLikeExpression(lhs, rhs)
 
-        let quantumDep =
+        let quantumDep = // NOTE: for an operation call we always need to set the quantum dependency to true
```

Comment:
Just for my own understanding, does this mean that every operation call is assumed to use quantum resources? That seems like it would make any use of the quantum dependency info rather pessimistic, but perhaps that is the itent.

Reply:
Yes, quantum dependency is pessimistic. It is used for generating adjoints, and in that sense the info merely captures if something can be dependent on side effects. So, this is basically the definition of quantum dependency - it has nothing to do with qubits, and instead merely tracks if something has a direct or indirect dependency on an operation call. I think the reason this was never an issue is likely that the syntax generator is primarily used for code for which functor support has already been generated (e.g. this is used for the generated functor specializations themselves).